### ISSUE TITLE
Speaker_id during inference

### ISSUE THREAD

**Issue by Srija616** [ylacombe/finetune-hf-vits#5]
Hi @ylacombe! I have a multi-speaker data using which I have trained the hindi checkpoint. I wanted to generate a particular speaker's voice during inference. Is there any way to do that using the inference code given in the README?

Here is how my current code looks:
`import scipy

from transformers import pipeline
import time
model_id = "./vits_finetuned_hindi"
synthesiser = pipeline("text-to-speech", model_id, device=0) # add device=0 if you want to use a GPU
speech = synthesiser("वहीं पंजाब सरकार ने सरबत खालसा के आयोजन के लिए, पंजाब के भठिंडा ज़िले में, तलवंडी साबो में, जगह देने से मना कर दिया है।")
scipy.io.wavfile.write("hindi_1.wav", rate=speech["sampling_rate"], data=speech["audio"][0])`

**Comment by ylacombe** [comment]
Hey @Srija616, you can use the [kwarg `speaker_id` like this](https://huggingface.co/docs/transformers/main/en/main_classes/pipelines#transformers.TextToAudioPipeline.__call__):

```python
forward_params = {"speaker_id": XXXX}
text = "वहीं पंजाब सरकार ने सरबत खालसा के आयोजन के लिए, पंजाब के भठिंडा ज़िले में, तलवंडी साबो में, जगह देने से मना कर दिया है।"

speech = synthetiser(text, forward_params=forward_params)
```

Did you finetune using the multi-speaker feature from the training code ?
Also, I'm quite curious about your feeling on the quality of the model, don't hesitate to let me know,
Best

**Comment by Srija616** [comment]
@ylacombe  Yes we have two speakers for Hindi (male, female)  and these are the two params we tweaked to enable multispeaker training. Just wondering if there are other params that need to be defined for multispeaker training.
![image](https://github.com/ylacombe/finetune-hf-vits/assets/33909898/da4c53a9-1f4f-4324-ad06-aa589d69053b)

We are also facing two issues:
1. During finetuning, the train_loss_kl and val_loss_kl are both going to infinity - tested it with the English finetuning using ylacombe/mms-tts-eng-train model and here too, we are facing the same problem. The train_loss_disc has NaN values and the mel loss for both train and validation are not converging. 
The synthesized sample although seems to good with pronunciation and naturalness for English. For Hindi, we have pronunciation and naturalness issues.

2. The speaker generated by the model is not similar to the speaker of the dataset, even though I passed the speaker_id as you mentioned in the previous comment.

Adding the wandb charts for our Hindi and English runs:
1. Hindi
![image](https://github.com/ylacombe/finetune-hf-vits/assets/33909898/e439ffb5-3506-4503-a019-141787fdab30)

2. English
![image](https://github.com/ylacombe/finetune-hf-vits/assets/33909898/59bbb8cd-4d85-4df6-a475-e544b9a25342)

@ylacombe Was wondering if you have some thoughts on why these losses are going to infinity or Nan. It is possible that we are missing something trivial.

I can share the generated samples over mail, if you'd like to here.

**Comment by ylacombe** [comment]
Hey @Srija616, sorry for the late response! Nice project here!
If you have two speakers, I'd recommend finetuning two different times, since the original model only had one speaker, so the speaker embeddings must be learn from scratch.

Can you send me your training config ? I do have some great finetuning results on a single speaker fine-tuning

**Comment by gsyllas** [comment]
Hello @ylacombe , as I am currently finetuning the mms_tts_ell on a single speaker dataset would it be possible to assist me with the training configurations? My dataset consists ~4 hourds.

**Comment by gpy1234** [comment]
Hi @Srija616 Did you solve the issue?

**Comment by Srija616** [comment]
Hi @

> Hey [@Srija616](https://github.com/Srija616), sorry for the late response! Nice project here! If you have two speakers, I'd recommend finetuning two different times, since the original model only had one speaker, so the speaker embeddings must be learn from scratch.
> 
> Can you send me your training config ? I do have some great finetuning results on a single speaker fine-tuning



Hi @ylacombe, unfortunately I did not get back to this code and did not attempt to fix it. We moved to training VITS from scratch.

**Comment by fiifinketia** [comment]
Hi @ylacombe, even though I passed the speaker_id as instructed, the voice produced by the model still doesn’t match the speaker in the dataset.
